aten/src/ATen/native/cuda/ScanKernels.cu

```diff
@@ -367,33 +366,6 @@ tensor_kernel_scan_innermost_dim(
       row_buf, tgt_, src_, num_rows, row_size, init, binary_op);
 }
 
-template <
-    typename T,
-    int num_threads_x,
-    int num_threads_y,
-    class BinaryFunction>
-__global__ typename std::enable_if<c10::is_complex_t<T>::value, void>::type
-tensor_kernel_scan_innermost_dim(
-    T* tgt_,
-    T* src_,
-    unsigned num_rows,
-    unsigned row_size,
-    T init,
-    BinaryFunction binary_op) {
-  // As we cannot directly initialize shared array for complex types
-  // Reference:
-  //  `error: initializer not allowed for __shared__ variable`
-  // We instead get the base scalar type and allocate twice number of
-  // elements required of base type and reinterpret them as complex.
-  using base_t = typename scalar_value_type<T>::type;
-  __shared__ base_t sbuf[num_threads_y][4 * num_threads_x];
```

Comment:
I don't think using `reinterpret_cast` for the unified path for complex and non-complex types is a good idea. The reason I feel this is because, `reinterpret_cast` is like throwing away the guarantees provided by the compiler and we have to ascertain them ourselves. So in ideal scenario, we should remove the `reinterpret_cast` from the complex as well.

Apologies for the delayed response.